### PR TITLE
Change user to principal in example schema

### PR DESCRIPTION
**Pull request by wscalf**
## Describe your changes

- ...

## Ticket reference (if applicable)
Fixes #

## Checklist

* [ ] Are the agreed upon acceptance criteria fulfilled?

* [ ] Was the 4-eye-principle applied? (async PR review, pairing, ensembling)

* [ ] Do your changes have passing automated tests and sufficient observability?

* [ ] Are the work steps you introduced repeatable by others, either through automation or documentation?
  * [ ] If automation is possible but not done due to other constraints, a ticket to the tech debt sprint is added
  * [ ] An SOP (Standard Operating Procedure) was created

* [ ] The Changes were automatically built, tested, and  - if needed, behind a feature flag - deployed to our production environment. (**Please check this when the new deployment is done and you could verify it.**)

* [ ] Are the agreed upon coding/architectural practices applied?

* [ ] Are security needs fullfilled? (e.g. no internal URL)

* [ ] Is the corresponding Ticket in the right state? (should be on "review" now, put to done when this change made it to production)

* [ ] For changes to the public API / code dependencies: Was the whole team (or a sufficient amount of ppl) able to review?

### CODE REVIEW

File: internal/data/spicedb_test.go [hunk 1]
```diff
@@ -93,27 +94,27 @@ func TestCreateRelationshipWithSubjectRelation(t *testing.T) {
...
 	container.WaitForQuantizationInterval()
 
-	exists := CheckForRelationship(spiceDbRepo, "bob", "rbac", "user", "", "member", "rbac", "group", "bob_club")
+	exists := CheckForRelationship(spiceDbRepo, "bob", "rbac", "principal", "", "member", "rbac", "group", "bob_club")
 	assert.True(t, exists)
 
 	exists = CheckForRelationship(spiceDbRepo, "bob_club", "rbac", "group", "member", "subject", "rbac", "role_binding", "fan_binding")
 	assert.True(t, exists)
 
 	// zed permission check rbac/role_binding:fan_binding subject rbac/user:bob
```
Comment (lennysgarage) [inline internal/data/spicedb_test.go:103]:
```suggestion
	// zed permission check rbac/role_binding:fan_binding subject rbac/principal:bob
```
Couple comments that still reference user

File: internal/data/spicedb_test.go [hunk 2]
```diff
@@ -93,27 +94,27 @@ func TestCreateRelationshipWithSubjectRelation(t *testing.T) {
...
 	// bob is a subject of fan_binding
-	runSpiceDBCheck(t, ctx, spiceDbRepo, "user", "rbac", "bob", "subject", "role_binding", "rbac", "fan_binding", apiV1beta1.CheckResponse_ALLOWED_TRUE)
+	runSpiceDBCheck(t, ctx, spiceDbRepo, "principal", "rbac", "bob", "subject", "role_binding", "rbac", "fan_binding", apiV1beta1.CheckResponse_ALLOWED_TRUE)
 
 	// zed permission check rbac/role_binding:fan_binding subject rbac/user:alice
```
Comment (lennysgarage) [inline internal/data/spicedb_test.go:107]: here

File: internal/data/spicedb_test.go [hunk 3]
```diff
@@ -93,27 +94,27 @@ func TestCreateRelationshipWithSubjectRelation(t *testing.T) {
...
 	// alice is NOT a subject of fan_binding
-	runSpiceDBCheck(t, ctx, spiceDbRepo, "user", "rbac", "alice", "subject", "role_binding", "rbac", "fan_binding", apiV1beta1.CheckResponse_ALLOWED_FALSE)
+	runSpiceDBCheck(t, ctx, spiceDbRepo, "principal", "rbac", "alice", "subject", "role_binding", "rbac", "fan_binding", apiV1beta1.CheckResponse_ALLOWED_FALSE)
 
-	// zed permission check rbac/role_binding:fan_binding view_the_thing rbac/user:bob
-	// bob has view_the_thing permission
-	runSpiceDBCheck(t, ctx, spiceDbRepo, "user", "rbac", "bob", "view_the_thing", "role_binding", "rbac", "fan_binding", apiV1beta1.CheckResponse_ALLOWED_TRUE)
+	// zed permission check rbac/role_binding:fan_binding view_widget rbac/user:bob
```
Comment (lennysgarage) [inline internal/data/spicedb_test.go:111]: here

File: internal/data/spicedb_test.go [hunk 4]
```diff
@@ -93,27 +94,27 @@ func TestCreateRelationshipWithSubjectRelation(t *testing.T) {
...
+	// bob has view_widget permission
+	runSpiceDBCheck(t, ctx, spiceDbRepo, "principal", "rbac", "bob", "view_widget", "role_binding", "rbac", "fan_binding", apiV1beta1.CheckResponse_ALLOWED_TRUE)
 
 	// zed permission check rbac/role_binding:fan_binding subject rbac/user:alice
```
Comment (lennysgarage) [inline internal/data/spicedb_test.go:115]: and here

File: internal/data/spicedb_test.go [hunk 5]
```diff
@@ -652,10 +653,10 @@ func TestSpiceDbRepository_CheckPermission(t *testing.T) {
 		},
 		Id: "test",
 	}
-	// zed permission check workspace:test view_the_thing user:bob --explain
+	// zed permission check workspace:test view_widget user:bob --explain
```
Comment (lennysgarage) [inline internal/data/spicedb_test.go:656]: here as well

File: internal/data/spicedb_test.go [hunk 6]
```diff
@@ -623,11 +624,11 @@ func TestSpiceDbRepository_CheckPermission(t *testing.T) {
 	//role_binding:rb_test#subject@user:bob
 	//role:rl1#view_the_thing@user:*
```
Comment (lennysgarage) [inline internal/data/spicedb_test.go:625]: 3 here

File: internal/data/spicedb_test.go [hunk 7]
```diff
@@ -677,17 +678,17 @@ func TestSpiceDbRepository_CheckPermission(t *testing.T) {
...
-			SubjectType:      pointerize("user"),
+			SubjectType:      pointerize("principal"),
 		},
 	})
 	if !assert.NoError(t, err) {
 		return
 	}
 
-	// zed permission check workspace:test view_the_thing user:bob --explain
+	// zed permission check workspace:test view_widget user:bob --explain
```
Comment (lennysgarage) [inline internal/data/spicedb_test.go:688]: and here

**Review by lennysgarage — CHANGES_REQUESTED** [review]
Looks great! Left a couple comments on places that still reference the `user` type that should be updated.

**Comment by wscalf** [comment]
Made a pass at updating comments. I think I covered the ones mentioned here and a few more, _but_, some may not be exactly right, and at least one I just removed. It looks like they've been out of sync with the schema for a while (missing namespace prefixes, t_-prefixes, etc)

**Review by lennysgarage — APPROVED** [review]
LGTM!

**Pull request merged**